src/main/kotlin/com/hibob/academy/fillters/AuthenticationFilter.kt

```diff
@@ -0,0 +1,33 @@
+package com.hibob.academy.fillters
+
+import com.hibob.academy.service.SessionService
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.Jwts
+import jakarta.ws.rs.container.ContainerRequestContext
+import jakarta.ws.rs.container.ContainerRequestFilter
+import jakarta.ws.rs.core.Cookie
+import org.springframework.stereotype.Component
+import jakarta.ws.rs.core.Response
+import jakarta.ws.rs.ext.Provider
+
+@Provider
+@Component
+class AuthenticationFilter(private val sessionService: SessionService) : ContainerRequestFilter {
+
+    override fun filter(requestContext: ContainerRequestContext) {
+        if (requestContext.uriInfo.path == "api/user_session") return
+        val jwtCookie: Cookie? = requestContext.cookies["jwt"]
+        val claims: Claims? = sessionService.verify(jwtCookie?.value)
```

Comment:
nice! i like that you concentrated all JWT logic in one class. well done

Reply:
maybe consider only calling the sessionService if the cookie isn't null. you can use the let function in kotlin:
```suggestion
        
        val claims = jwtCookie?.let { sessionService.verify(it.value) }
```